src/components/section/Dialog.tsx

```diff
@@ -259,3 +259,98 @@ export const DeleteDialog: React.FC<DeleteDialogProps> = ({
     </Transition>
   );
 };
+
+interface LeaveDialogProps {
+  isOpen: boolean;
+  close: () => void;
+  eventName: string;
+  onSubmit: () => Promise<void>;
+}
+
+export const LeaveDialog: React.FC<LeaveDialogProps> = ({
+  isOpen,
+  close,
+  eventName,
+  onSubmit,
+}) => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  return (
+    <Transition show={isOpen} as={Fragment}>
+      <Dialog
+        as="div"
+        className="fixed inset-0 z-10 overflow-y-auto"
+        onClose={() => close()}
+      >
+        <div className="min-h-screen px-4 text-center">
+          <Transition.Child
+            as={Fragment}
+            enter="ease-out duration-300"
+            enterFrom="opacity-0"
+            enterTo="opacity-100"
+            leave="ease-in duration-200"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+          >
+            <Dialog.Overlay className="fixed inset-0 bg-black bg-opacity-50" />
+          </Transition.Child>
+          <span
+            className="inline-block h-screen align-middle"
+            aria-hidden="true"
+          ></span>
+          <Transition.Child
+            as={Fragment}
+            enter="ease-out duration-300"
+            enterFrom="opacity-0 scale-95"
+            enterTo="opacity-100 scale-100"
+            leave="ease-in duration-200"
+            leaveFrom="opacity-100 scale-100"
+            leaveTo="opacity-0 scale-95"
+          >
+            <div className="my-space-x-8 inline-block w-[450px] max-w-lg transform rounded-xl bg-white p-6 shadow-xl transition-all">
+              <Dialog.Title className="text-left text-lg font-semibold leading-6 text-gray-900">
+                Confirm Leave Event
+              </Dialog.Title>
+              <div className="mt-4 text-left text-2xl font-bold">
+                {eventName}
+              </div>
+              <div className="mt-2 text-left text-xs text-gray-500">
+                You will leave the event permanently, including all availability
+                data. This action is irreversible and can not be undone.
+              </div>
```

Comment:
We should probably inform the user that the time slots will be deleted as well, and prob make this action more informative with event name included. prob sth like `You will leave <event name> permanently, your selected time slots will also be deleted. This action is irreversible and can not be undone.`

We could also add a typing check like when you delete projects on Vercel, but I guess we don't have to do that for participants leaving events but we can consider doing this for when owner deletes the whole event